src/JustEat.StatsD/TimerExtensions.cs

```diff
@@ -5,7 +5,7 @@ namespace JustEat.StatsD
 {
     public static class TimerExtensions
     {
-        public static IDisposable StartTimer(this IStatsDPublisher publisher, string bucket)
+        public static IDisposableTimer StartTimer(this IStatsDPublisher publisher, string bucket = null)
```

Comment:
So the idea behind starting with a null/empty string is to allow for the case of "I am going to set the stat name in entirety later (e.g. depending on operation success or failure), so there's no point in putting anything in now".

However you are right, it does make the default case an unwelcome one - this does not encourage the dev to "fall into the pit of success" where the stat has a name.

So I will
- raise an exception if the bucket is null at the end
- possibly also revert this line and insist on a string to start off with. The dev can still specify null explicitly at start, but this is more visible.